NeoML/include/NeoML/Dnn/DnnSimpleAPI.h

```diff
@@ -0,0 +1,521 @@
+
+#pragma once
+
+#include <NeoML/Dnn/DnnLambdaHolder.h>
+#include <NeoML/NeoMLDefs.h>
+#include <NeoML/Dnn/Dnn.h>
+#include <NeoML/Dnn/Layers/DropoutLayer.h>
+#include <NeoML/Dnn/Layers/ActivationLayers.h>
+#include <NeoML/Dnn/Layers/GELULayer.h>
+#include <NeoML/Dnn/Layers/FullyConnectedLayer.h>
+#include <NeoML/Dnn/Layers/LossLayer.h>
+#include <NeoML/Dnn/Layers/SourceLayer.h>
+#include <NeoML/Dnn/Layers/SinkLayer.h>
+#include <NeoML/Dnn/Layers/PositionalEmbeddingLayer.h>
+#include <NeoML/Dnn/Layers/SoftmaxLayer.h>
+#include <NeoML/Dnn/Layers/AddToObjectLayer.h>
+#include <NeoML/Dnn/Layers/TransformLayer.h>
+#include <NeoML/Dnn/Layers/TransposeLayer.h>
+#include <NeoML/Dnn/Layers/ConcatLayer.h>
+#include <NeoML/Dnn/Layers/BatchNormalizationLayer.h>
+#include <NeoML/Dnn/Layers/ObjectNormalizationLayer.h>
+#include <NeoML/Dnn/Layers/MultiheadAttentionLayer.h>
+#include <NeoML/Dnn/Layers/MatrixMultiplicationLayer.h>
+#include <NeoML/Dnn/Layers/EltwiseLayer.h>
+#include <NeoML/Dnn/Layers/GlobalMaxPoolingLayer.h>
+#include <NeoML/Dnn/Layers/GlobalMeanPoolingLayer.h>
+#include <NeoML/Dnn/Layers/LstmLayer.h>
+#include <NeoML/Dnn/Layers/GruLayer.h>
+#include <NeoML/Dnn/Layers/SubSequenceLayer.h>
+#include <NeoML/Dnn/Layers/ArgmaxLayer.h>
+#include <NeoML/Dnn/Layers/MultichannelLookupLayer.h>
+#include <NeoML/Dnn/Layers/ConvLayer.h>
+#include <NeoML/Dnn/Layers/MaxOverTimePoolingLayer.h>
+#include <NeoML/Dnn/Layers/3dConvLayer.h>
+#include <NeoML/Dnn/Layers/3dPoolingLayer.h>
+#include <NeoML/Dnn/Layers/TransposedConvLayer.h>
+#include <NeoML/Dnn/Layers/AccumulativeLookupLayer.h>
+#include <NeoML/Dnn/Layers/AccuracyLayer.h>
+#include <NeoML/Dnn/Layers/AttentionLayer.h>
+#include <NeoML/Dnn/Layers/BinaryFocalLossLayer.h>
+#include <NeoML/Dnn/Layers/CenterLossLayer.h>
+#include <NeoML/Dnn/Layers/ChannelwiseConvLayer.h>
+#include <NeoML/Dnn/Layers/CrfLayer.h>
+#include <NeoML/Dnn/Layers/CtcLayer.h>
+#include <NeoML/Dnn/Layers/DotProductLayer.h>
+#include <NeoML/Dnn/Layers/EnumBinarizationLayer.h>
+#include <NeoML/Dnn/Layers/FocalLossLayer.h>
+#include <NeoML/Dnn/Layers/FullyConnectedSourceLayer.h>
+#include <NeoML/Dnn/Layers/ImageAndPixelConversionLayer.h>
+#include <NeoML/Dnn/Layers/ImageResizeLayer.h>
+#include <NeoML/Dnn/Layers/MaxOverTimePoolingLayer.h>
+#include <NeoML/Dnn/Layers/MultiHingeLossLayer.h>
+#include <NeoML/Dnn/Layers/PrecisionRecallLayer.h>
+#include <NeoML/Dnn/Layers/QualityControlLayer.h>
+#include <NeoML/Dnn/Layers/ReorgLayer.h>
+#include <NeoML/Dnn/Layers/RepeatSequenceLayer.h>
+#include <NeoML/Dnn/Layers/SequenceSumLayer.h>
+#include <NeoML/Dnn/Layers/SequenceSumLayer.h>
+#include <NeoML/Dnn/Layers/SplitLayer.h>
+#include <NeoML/Dnn/Layers/TimeConvLayer.h>
+#include <NeoML/Dnn/Layers/TransposedConvLayer.h>
+#include <NeoML/Dnn/Layers/Upsampling2DLayer.h>
+
+namespace NeoML {
+
+// Layer output.
+class NEOML_API CLayerOutput {

```

Comment:
It actually describes a layer input) There is the same struct in CDnn - "CDnnLayerLink", we should combine them!

Reply:
Ok, but what is the best place to locate this CDnnLayerLink?

Reply:
I think, Dnn.h.